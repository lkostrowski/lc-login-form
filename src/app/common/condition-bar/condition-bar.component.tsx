import cx from 'classnames';
import React, { HTMLProps } from 'react';

import styles from './condition-bar.module.scss';

interface Props extends HTMLProps<HTMLDivElement> {
    state: 'valid' | 'invalid' | undefined;
    label: string;
}

export const ConditionBar = ({ state, label, className, ...props }: Props) => {
    return (
        <div className={cx(className, styles.container)} {...props}>
            <div
                className={cx(styles.bar, {
                    [styles.invalid]: state === 'invalid',
                    [styles.valid]: state === 'valid',
                })}
            />
            <span
                className={cx(styles.label, {
                    [styles.invalid]: state === 'invalid',
                    [styles.valid]: state === 'valid',
                })}
            >
                {label}
            </span>
        </div>
    );
};
