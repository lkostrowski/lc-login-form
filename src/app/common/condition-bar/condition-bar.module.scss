@import '../../../theme/colors';
@import '../../../theme/vars';

.container {
}

.bar {
    height: 3px;
    border-radius: $borderRadius;
    background: $lightGray;
    margin-bottom: 0.5rem;

    &.invalid {
        background: $red;
    }

    &.valid {
        background: $green;
    }
}

.label {
    color: $lightGray;
    display: block;
    text-transform: uppercase;
    font-weight: 500;
    font-size: 1rem;
    white-space: nowrap;
    text-align: center;

    &.invalid {
        color: $red;
    }

    &.valid {
        color: $green;
    }
}
