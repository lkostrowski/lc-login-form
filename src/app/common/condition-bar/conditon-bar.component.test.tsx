import { cleanup, render } from '@testing-library/react';
import React from 'react';

import { ConditionBar } from './condition-bar.component';

afterEach(cleanup);

describe('Condition Bar component', () => {
    describe('Snapshots', () => {
        test('Invalid', () => {
            const { container } = render(
                <ConditionBar label='Some condition' state='invalid' />,
            );

            expect(container).toMatchSnapshot();
        });

        test('Valid', () => {
            const { container } = render(
                <ConditionBar label='Some condition' state='valid' />,
            );

            expect(container).toMatchSnapshot();
        });
    });
});
