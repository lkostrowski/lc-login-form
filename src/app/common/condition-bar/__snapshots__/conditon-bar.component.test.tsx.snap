// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Condition Bar component Snapshots Invalid 1`] = `
<div>
  <div
    class="container"
  >
    <div
      class="bar invalid"
    />
    <span
      class="label invalid"
    >
      Some condition
    </span>
  </div>
</div>
`;

exports[`Condition Bar component Snapshots Valid 1`] = `
<div>
  <div
    class="container"
  >
    <div
      class="bar valid"
    />
    <span
      class="label valid"
    >
      Some condition
    </span>
  </div>
</div>
`;
